### ISSUE TITLE
File upload

### ISSUE THREAD

**Issue by fizerkhan** [icebob/vue-form-generator#172]
I want to upload CSV file. What is the input type for file upload?

**Comment by cristijora** [comment]
I don't see file upload being handled in the code. You can go with your own [custom component](https://icebob.gitbooks.io/vueformgenerator/content/fields/custom_fields.html) and use any of the exisiting file upload components that are out there (e.g [here](https://github.com/lian-yue/vue-upload-component) )

**Comment by fizerkhan** [comment]
@cristijora Your commit will fix the issue right?

**Comment by cristijora** [comment]
Yes, the [PR](https://github.com/icebob/vue-form-generator/pull/173) got merged and it will be available in the next build or if you need it now simply use the [js from here](https://github.com/icebob/vue-form-generator/tree/master/dist)

**Comment by fizerkhan** [comment]
@cristijora Thanks. 👍

**Comment by fizerkhan** [comment]
@cristijora I got this error after integrating this js file

```
DOMException: Failed to set the 'value' property on 'HTMLInputElement': This input element accepts a filename, which may only be programmatically set to the empty string.
    at Array.updateDOMProps (http://localhost:3000/app/app.js:10043:19)
    at patchVnode (http://localhost:3000/app/app.js:9559:62)
    at updateChildren (http://localhost:3000/app/app.js:9480:9)
    at patchVnode (http://localhost:3000/app/app.js:9564:29)
    at VueComponent.patch [as __patch__] (http://localhost:3000/app/app.js:9689:9)
    at VueComponent.Vue._update (http://localhost:3000/app/app.js:7663:19)
    at VueComponent.<anonymous> (http://localhost:3000/app/app.js:7630:10)
    at Watcher.get (http://localhost:3000/app/app.js:6827:27)
    at Watcher.run (http://localhost:3000/app/app.js:6896:22)
    at flushSchedulerQueue (http://localhost:3000/app/app.js:6710:13)
logError @ app.js:5601
```

**Comment by cristijora** [comment]
I will look into it.

**Comment by cristijora** [comment]
@fizerkhan Try not to set the initial model value for the file input to an empty string. Does this appear every time you upload something or only initially?

**Comment by fizerkhan** [comment]
@cristijora It happens only after I select a file from the file dialog.

**Comment by fizerkhan** [comment]
@cristijora I fixed this by setting file value to ''.

```
this.model.file = '';
```

**Comment by BramVer** [comment]
Is anyone still looking into this? Just wanted to bump the thread, since I'm experiencing the same annoyance.  
Fizerkhan's fix didn't solve it for me, whether I put the initial value to null or an empty string.   

The solution works, and files get transferred flawlessly, only the warning is something I'd like to omit in my final product.

**Comment by icebob** [comment]
What kind of warning?

**Comment by BramVer** [comment]
@icebob The same warning fizerkhan received. It's a DOMException, but nothing crashes and the file selection works all the same.

![image](https://cloud.githubusercontent.com/assets/16743200/24947769/52f71e40-1f68-11e7-9771-ed33f1ffa0db.png)

In my model I tried setting initial value to an empty string and null, but that doesn't make any difference.   
My schema looks like this:
```javascript

          {
            //ATTACHMENT
            type: "input",
            inputType: "file",
            model: "attachments",
            label: "Attachment",
            files: true,
            multiple: true,

            styleClasses: 'col-md-6',
          },
```

It happens right after you confirm which file to upload in the popup window.

**Comment by fizerkhan** [comment]
@BramVer Can you please set empty string `''` for other input fields in the form?

**Comment by BramVer** [comment]
@fizerkhan You meant setting all values in the model to an empty string, right?

That doesn't do it for me sadly, besides getting the same warning, other input fields dont work because they're not initialized of course.

I see that you manually set this.model.file to an empty string, can I ask where you did this? I tried manually setting that in 'created' as well, but does nothing.

**Comment by fizerkhan** [comment]
@BramVer I did like this

```
this.model = schemaUtils.createDefaultObject(mySchema);
this.model.file = '';
this.model.content = '';
```

Even after setting `this.model.file = '';`, I got the same warning for `content` input. So I set it to empty too.

**Comment by BramVer** [comment]
@fizerkhan This removed the warning, thank you. 

Instead of doing this (which is more in line with the structure of the code):
```javascript
      name: '...',

      model: {
        start_date: moment(),
        end_date: moment().add(1, 'days'),
        description: "",
        leave_type: null,
        attachments: null,
      },

      schema: { ... }
```
I now do this:
```javascript
  created: function() {
    this.model = VueFormGenerator.schema.createDefaultObject(this.schema);

    this.model.start_date = moment();
    this.model.end_date = moment().add(1, 'days');
    this.model.attachments = '';
  },

```

I still don't think this should be seen as a permanent solution, but more as a workaround,
since it doesn't fit well with the flow of the remaining code.
If there're no plans on improving this, perhaps mention this issue in the docs as a workaround.

**Comment by fizerkhan** [comment]
@BramVer Great.
I agree with you that it is definitely a Work ground. It must be fixed or documented somewhere else.

**Comment by nthorpe** [comment]
Hi @cristijora. Has this issue been fixed? I'm receiving the same error as https://github.com/icebob/vue-form-generator/issues/172#issuecomment-293506515 using 2.0.0-beta8

**Comment by cristijora** [comment]
Not very sure whether it has been fixed. I did not follow the thread
unfortunately. Detected the issue and made a PR.
Maybe @icebob can provide more details here

**Comment by icebob** [comment]
Released in [v2.1.1](https://github.com/icebob/vue-form-generator/releases/tag/v2.1.1)
